### ISSUE TITLE
Problem to get reward from events

### ISSUE THREAD

**Issue by MaryLightning** [kakaroto/SWProxy#172]
Hello,

i use swproxy some weeks. I have problems to get reward from event if i must it collect. I press on button to collect it, and i get an error after one minutes.

Errormessage on ingame eventpage: "A temporary error has occurred. Please try again later.(*error)"

If i go in the game without swproxy, then i get the reward after a second.

A other problem with events: I can not enter if event have not rewards to collect it.

I have a workaround for both of problems. This Server "m-mercury.qpyou.cn" enter in the field "Bypass proxy for".

**Comment by Viper2k4** [comment]
Thats relativ common. SWProxy it self ist a Proxy. So after u set him all traffic run to the proxy. So u will got even more error (download items on itunes and so on). If u want stay connect to the proxy all the time i would use an pac-file like this:

```
function FindProxyForURL(url, host) {
    if (shExpMatch(host, "summonerswar*qpyou.cn"))
        return "PROXY 192.168.1.18:8080;DIRECT";

    return "DIRECT";
}
```

So only the API-Call will run to the proxy and the rest is running direct to the internet.

**Comment by kakaroto** [comment]
@Viper2k4 well, technically the proxy shouldn't prevent those other sites from working, it's a proxy, it's meant to proxy stuff out and it should just work.
@MaryLightning Clearly there's a bug preventing the proxy from working properly, but it's not meant to be used in that way. SWProxy is meant to be used only for a short amount of time, such as :
* Open SWProxy
* set your proxy on device
* Open Summoners War
* Monsters & runes export
* Unset the proxy from your device
* Then continue playing....

It was never really meant to be kept open while people are playing. While I know there are plugins that require SWProxy to keep running, those plugins are not officially supported.
I'm keeping this issue open though in case someone volunteers to fix it.

**Comment by Viper2k4** [comment]
@kakaroto yes it should be. But hands down the proxy.py is not the best proxy out there (but for this purpose truly a good one). And i can 100% proof that many https parts (itunes, some internet-radios and so on) not working as intended.

I use this projekt + SwarfarmLogger.py (from the other projekt). And run the proxy 24/7 and is running just perfect (combined with the pac above).

**Comment by kakaroto** [comment]
yeah, I know, proxy.py isn't great but I googled "proxy python" and probably clicked on the first link, or at least, the first one I found which allowed us to intercept the proxy data with an API. I know there are many other proxy implementation, and probably some much better/stabler ones out there (I believe there's one in twister?) but I just lack the motivation to continue working on this project and to rewrite it.

**Comment by Viper2k4** [comment]
Yes i feel the same. But i'm atm. on an other time consuming projekt. And the actual version runs just fine with the pac. So i'm also lack a bit motivation to switch the proxy^^

But twisted seems nice. I test it and it work just well (just test it on some web-calls).

**Comment by JasonGoemaat** [comment]
@Viper2k4 you've earned my undying gratitude.  For anyone else that wants to do this just create a file with a javascript function like Viper2k4 suggested and make it available to your phone's browser somehow.  I just put it in my iis root folder on my home computer I'm using for a proxy.  In my iphone then I just went to auto proxy configuration and gave a url to access that file and voila!

**Comment by kakaroto** [comment]
Hummm... maybe SWProxy could auto  generate that file and use a small python web server to provide it. Then the UI could just give the users the URL to enter for the auto settings file  instead of the ip+port.
